### ISSUE TITLE
create-domain-job.sh: line 4: $'\r': command not found

### ISSUE THREAD

**Issue by gwbatte** [oracle/weblogic-kubernetes-operator#2636]
HiHi,

Following > Samples > Azure Kubernetes Service > Domain home on a PV:
[]((https://oracle.github.io/weblogic-kubernetes-operator/samples/azure-kubernetes-service/domain-on-pv/))

Step: Asking Operator to create WebLogic Server domain within the Azure: AKS Cluster.

$ ./create-domain.sh -i ~/azure/weblogic-on-aks/domain1.yaml -o ~/azure_11_22_2021 -e -v
Input parameters being used
export version="create-weblogic-sample-domain-inputs-v1"
export adminPort="7001"
export adminServerName="admin-server"
export domainUID="domain1"
export domainHome="/shared/domains/domain1"
export serverStartPolicy="IF_NEEDED"
export clusterName="cluster-1"
export configuredManagedServerCount="5"
export initialManagedServerReplicas="2"
export managedServerNameBase="managed-server"
export managedServerPort="8001"
export image="<ACR_Name>.azurecr.io/middleware/weblogic:12.2.1.4"
export imagePullPolicy="Always"
export imagePullSecretName="wlsregcred"
export productionModeEnabled="true"
export weblogicCredentialsSecretName="domain1-weblogic-credentials"
export includeServerOutInPodLog="true"
export logHome="/shared/logs/domain1"
export httpAccessLogInLogHome="true"
export t3ChannelPort="30012"
export exposeAdminT3Channel="false"
export adminNodePort="30701"
export exposeAdminNodePort="true"
export namespace="idam"
javaOptions=-Dweblogic.StdoutDebugEnabled=false -XX:MinRAMPercentage=25.0 -XX:MaxRAMPercentage=50.0
export persistentVolumeClaimName="wls-azurefile-1637114823"
export domainPVMountPath="/shared"
export createDomainScriptsMountPath="/u01/weblogic"
export createDomainScriptName="create-domain-job.sh"
export createDomainFilesDir="wlst"
export serverPodMemoryRequest="1.5Gi"
export serverPodCpuRequest="250m"
export istioEnabled="false"
export istioReadinessPort="8888"

createFiles - valuesInputFile is /c/Users/<userid>/azure/weblogic-on-aks/domain1.yaml
createDomainScriptName is create-domain-job.sh
Generating /c/Users/<userid>/azure_11_22_2021/weblogic-domains/domain1/create-domain-job.yaml
Generating /c/Users/<userid>/azure_11_22_2021/weblogic-domains/domain1/delete-domain-job.yaml
Generating /c/Users/<userid>/azure_11_22_2021/weblogic-domains/domain1/domain.yaml
/c/Users/<userid>/11-16-2021/weblogic-kubernetes-operator/kubernetes/samples/scripts/create-weblogic-domain/domain-home-on-pv/../../common/utility.sh: line 614: [[: MINGW64_NT-10.0-19042: syntax error: invalid arithmetic operator (error token is ".0-19042")
Checking to see if the secret domain1-weblogic-credentials exists in namespace idam
Checking if the persistent volume claim wls-azurefile-1637114823 in NameSpace idam exists
The persistent volume claim wls-azurefile-1637114823 already exists in NameSpace idam
/c/Users/<userid>/11-16-2021/weblogic-kubernetes-operator/kubernetes/samples/scripts/create-weblogic-domain/domain-home-on-pv/../../common/utility.sh: line 693: [[: MINGW64_NT-10.0-19042: syntax error: invalid arithmetic operator (error token is ".0-19042")
W1122 08:31:03.141965   26216 helpers.go:555] --dry-run is deprecated and can be replaced with --dry-run=client.
configmap/domain1-create-weblogic-sample-domain-job-cm configured
Checking the configmap domain1-create-weblogic-sample-domain-job-cm was created
'weblogic.domainUID' already has a value (domain1), and --overwrite is false
'weblogic.domainName' already has a value (domain1), and --overwrite is false
Checking if object type job with name domain1-create-weblogic-sample-domain-job exists
No resources found in idam namespace.
Creating the domain by creating the job /c/Users/<userid>/azure_11_22_2021/weblogic-domains/domain1/create-domain-job.yaml
job.batch/domain1-create-weblogic-sample-domain-job created
Waiting for the job to complete...
Found 3 pods, using pod/domain1-create-weblogic-sample-domain-job-tz279
status on iteration 1 of 30

Operator 3.3.4 [container-registry.oracle.com/middleware/weblogic-kubernetes-operator:3.3.4]
OIG Image used: [ Oracle Identity and Access Management 12.2.1.4 Products with Kubernetes (Doc ID 2723908.1) ]

kubectl logs domain1-create-weblogic-sample-domain-job-g48rc -n idam
/u01/weblogic/create-domain-job.sh: line 4: $'\r': command not found
/u01/weblogic/create-domain-job.sh: line 5: syntax error near unexpected token `$'{\r''
'u01/weblogic/create-domain-job.sh: line 5: `function exitIfError {

Any help would be much appreciated.

**Comment by tbarnes-us** [comment]
The WKO team is looking at this. In the mean-time, @galiacheng @edburns any suggestions for a work-around?

**Comment by tbarnes-us** [comment]
Is it possible the filer is not using WSL for Windows 10 (https://docs.microsoft.com/windows/wsl/install-win10)?

**Comment by edburns** [comment]
Internal issue: https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1441298

**Comment by galiacheng** [comment]
Hello @gwbatte we have to make sure all the shell script in `kubernetes/samples/scripts` are [LF ending](https://stackoverflow.com/questions/1552749/difference-between-cr-lf-lf-and-cr-line-break-types). WSL checkout the source code as CRLF by default. 

Please follow the steps to create WLS domain again.

#### Make sure scripts are LF ending 
There are several options to change the ending format, you can apply one of the following options.
- use [dos2unix](https://www.howtoinstall.me/ubuntu/18-04/dos2unix/)
   Install [dos2unix](https://www.howtoinstall.me/ubuntu/18-04/dos2unix/) and modify the script endings.
   ```
   sudo apt update
   sudo apt install dos2unix

   cd kubernetes/samples/scripts
   find . -type f -exec dos2unix {} \;
   ```
   After running the command, all shell script in samples folder should be LF ending.

- re-clone the source code
   Remove the previous source code, set `core.autocrlf` to `false` and clone again.

   ```
   git config --global core.autocrlf false
   git clone --branch v3.3.4 https://github.com/oracle/weblogic-kubernetes-operator.git
   ```

#### Delete the failure domain

As you ran the script to create WLS domain, there should be an existing domain, we can not override it directly. 
Run the following command to delete it.

```
kubectl delete domain domain1
kubectl delete configmap domain1-create-weblogic-sample-domain-job-cm
```

It's not allowed to create a domain in the existing domain path, so we have to remove it or create the domain in the second path. Here introduce how to re-create the storage file share.

Go to Azure portal -> your resource group -> your storage account - File shares 
Delete `wls-weblogic-1637114823`. The one you created before.
Add File share `wls-weblogic-1637114823`. Make the name is correct.
After the operation finishes, create the domain again.

#### Create domain

```
cd kubernetes/samples/scripts/create-weblogic-domain/domain-home-on-pv
./create-domain.sh -i ~/azure/weblogic-on-aks/domain1.yaml -o ~/azure_11_22_2021 -e -v
```

#### More

Microsoft Java EE on Azure team and Oracle WebLogic team provide Azure marketplace offer that makes it easy to get started with WebLogic Server on Azure, it may help you get started quickly, here list the documents
- Offer: https://ms.portal.azure.com/#create/oracle.20210620-wls-on-aks20210620-wls-on-aks
- Doc: https://oracle.github.io/weblogic-kubernetes-operator/userguide/aks/
- https://techcommunity.microsoft.com/t5/apps-on-azure/run-oracle-weblogic-server-on-azure-kubernetes-service/ba-p/2901524

Feel free to let me know your progress.

**Comment by rjeberhard** [comment]
Resolved with 3.3.5.

**Comment by gwbatte** [comment]
Thank you very much. I appreciate the quick resolution.
